pkg/util/util.go

```diff
@@ -163,6 +163,26 @@ func GenLogFields(costTime time.Duration, info *ProcessInfo, needTruncateSQL boo
 	logFields = append(logFields, zap.Uint64("txn_start_ts", info.CurTxnStartTS))
 	if memTracker := info.MemTracker; memTracker != nil {
 		logFields = append(logFields, zap.String("mem_max", fmt.Sprintf("%d Bytes (%v)", memTracker.MaxConsumed(), memTracker.FormatBytes(memTracker.MaxConsumed()))))
+
+		totalMemory := memTracker.MaxConsumed()
+		if totalMemory > 0 {
+			var buf strings.Builder
+			firstComma := false
+			for label, tracker := range memTracker.GetChildrenForTest() {
+				usage := tracker.MaxConsumed()
+				percentage := (float64(usage) / float64(totalMemory)) * 100
+				if percentage > 10 { // Only log if the usage is greater than 10%
+					if firstComma {
+						buf.WriteString(", ")
+					}
+					buf.WriteString(fmt.Sprintf("%d: %d Bytes (%.2f%%)", label, usage, percentage))
```

Comment:
currently the `label` is just the slice index of `GetChildrenForTest`. Do you mean `tracker.Label()`? And I'm not sure how can you use the output because it's not very meaningful.

Reply:
ho @lance6716 
Yes that is `label` of tracer.

In this part of logs::> `[mem_top_contributors="-28: 8192 Bytes (97.43%)"]`

One will see the top memory contributors (i.e. those that contribute >10%), if there are more than 1 processes, then all those processes will be shown here (in my case I only have 1 process that contribute >10% so only 1 is displayed)

I think getting this info can be helpful compared to existing logs we are getting

Reply:
I'm not sure how you can use `-28: 8192 Bytes (97.43%)`. `-28` does not have a clear meaning. I checked the code it's `LabelForMemDB`, do you think it's better we convert this number to its name?

And is it enough that only printing only one level of children? The original `memTracker` has a `String` method to display all levels.

Reply:
Ok I get the point....

So how should I go about this issue now?
Should I change the impl entirely or do some modification?

Can you help in this regard?

Reply:
can you take a look since you opened the issue? @ekexium